OrbitQt/orbitmainwindow.cpp

```diff
@@ -319,6 +310,145 @@ void OrbitMainWindow::SetupCodeView() {
   OrbitCodeEditor::setFileMappingWidget(ui->FileMappingWidget);
 }
 
+void OrbitMainWindow::SaveCurrentTabLayoutAsDefaultInMemory() {
+  default_tab_layout_.clear();
+  std::array<QTabWidget*, 2> tab_widgets = {ui->MainTabWidget, ui->RightTabWidget};
+  for (QTabWidget* tab_widget : tab_widgets) {
+    TabWidgetLayout layout;
+    for (int i = 0; i < tab_widget->count(); ++i) {
+      layout.tabs_and_titles.push_back(
+          std::make_pair(tab_widget->widget(i), tab_widget->tabText(i)));
+    }
+    layout.current_index = tab_widget->currentIndex();
+    default_tab_layout_[tab_widget] = layout;
+  }
+}
+
+void OrbitMainWindow::CreateTabBarContextMenu(QTabWidget* tab_widget, int tab_index,
+                                              const QPoint pos) {
+  QMenu context_menu(this);
+  QAction move_action;
+  QTabWidget* other_widget;
+
+  if (tab_widget == ui->MainTabWidget) {
+    move_action.setIcon(icon_keyboard_arrow_right_);
+    move_action.setText(QString("Move \"") + tab_widget->tabText(tab_index) + "\" to right pane");
+    other_widget = ui->RightTabWidget;
+  } else if (tab_widget == ui->RightTabWidget) {
+    move_action.setIcon(icon_keyboard_arrow_left_);
+    move_action.setText(QString("Move \"") + tab_widget->tabText(tab_index) + "\" to left pane");
+    other_widget = ui->MainTabWidget;
+  } else {
+    UNREACHABLE();
+  }
+
+  move_action.setEnabled(tab_widget->count() > 0);
+
+  QObject::connect(&move_action, &QAction::triggered, [this, tab_widget, other_widget, tab_index] {
+    QWidget* tab = tab_widget->widget(tab_index);
+    QString text = tab_widget->tabText(tab_index);
+    tab_widget->removeTab(tab_index);
+    other_widget->addTab(tab, text);
+    UpdateCaptureStateDependentWidgets();
+    if (tab->isEnabled()) {
+      other_widget->setCurrentWidget(tab);
+    }
+  });
+  context_menu.addAction(&move_action);
+  context_menu.exec(pos);
+}
+
+void OrbitMainWindow::UpdateCaptureStateDependentWidgets() {
+  auto SetTabEnabled = [this](QWidget* widget, bool enabled) -> void {
+    QTabWidget* tab_widget = FindParentTabWidget(widget);
+    CHECK(tab_widget != nullptr);
+    tab_widget->setTabEnabled(tab_widget->indexOf(widget), enabled);
+  };
+
+  const bool has_data = GOrbitApp->HasCaptureData();
+  const bool has_selection = has_data && GOrbitApp->HasSampleSelection();
+  const bool is_connected = GOrbitApp->IsConnectedToInstance();
+  const bool is_capturing = GOrbitApp->IsCapturing();
+
+  SetTabEnabled(ui->HomeTab, !is_capturing);
```

Comment:
So it's no longer possible to switch to the "home" tab altogether, right? The point of only disabling the content was to still show CPU usage while capturing. I remember a somewhat long discussion on this when we decided to disable the content of the "Home" tab only rather than the tab itself. But I guess we are moving away from that anyway.

Reply:
Ah, I see, makes sense. This function needs to be changed / merged when Anton's changes come in anyways, but I've changed it to reflect the previous behaviour.